### PR TITLE
Implement missing last interacted slot property in chiseled bookshelf

### PR DESCRIPTION
**Pull request by ipad54**
## Introduction
This PR implements missing `last interacted slot` property that is used in vanilla worlds for redstone power. Without this PR chiseled bookshelves loaded from vanilla worlds will discard their redstone power level.

Source: https://minecraft.wiki/w/Bedrock_Edition_level_format/Block_entity_format#Chiseled_Bookshelf

### API changes
Added the following methods:
- `ChiseledBookshelf::getLastInteractedSlot() : ?ChiseledBookshelfSlot`
- `ChiseledBookshelf::setLastInteractedSlot(?ChiseledBookshelfSlot $lastInteractedSlot) : self`

## Tests
Since PM doesn't have redstone implemented, I added some debug messages to the chat locally to test this.

Screenshot_20240901_120441
Screenshot_20240901_120339

### CODE REVIEW

**Review by IvanCraft623 — DISMISSED** [review]
LGTM, thanks for your constribution!

File: src/block/ChiseledBookshelf.php [hunk 1]
```diff
@@ -48,11 +48,30 @@ class ChiseledBookshelf extends Opaque{
...
 	protected function describeBlockOnlyState(RuntimeDataDescriber $w) : void{
 		$w->horizontalFacing($this->facing);
 		$w->enumSet($this->slots, ChiseledBookshelfSlot::cases());
 	}
 
+	public function readStateFromWorld() : Block{
+		$tile = $this->position->getWorld()->getTile($this->position);
+		if($tile instanceof TileChiseledBookshelf){
+			$this->lastInteractedSlot = $tile->getLastInteractedSlot();
+		}
```
Comment (dktapps) [inline src/block/ChiseledBookshelf.php:64]: This should have an `else` to set the property to `null` if the tile is missing

Comment (ipad54) [reply src/block/ChiseledBookshelf.php:64]: This property is initialized to null by default. It isn't enough?

Comment (dktapps) [reply src/block/ChiseledBookshelf.php:64]: On closer examination it appears some other blocks do the same thing you've done. Probably ought to be fixed, but that's for another time.

In core code this is only called in `getBlockAt()`, but there's no telling what plugins might do. There's no guarantee that the property is `null`.

**Review by dktapps — APPROVED** [review]

**Review by IvanCraft623 — APPROVED** [review]

**Pull request merged**